std/random.d

```diff
@@ -1807,6 +1807,90 @@ F[] uniformDistribution(F = double)(size_t n, F[] useThis = null)
 }
 
 /**
+Returns a random, uniformly chosen, element $(D e) from the supplied
+$(D Range range). If no random number generator is passed, the default
+$(D rndGen) is used.
+
+Params:
+    range = a random access range that has the length property defined
+    urng = (optional) random number generator to use;
+           if not specified, defaults to $(D rndGen)
+
+Returns:
+    A single random element drawn from the $(D range). If it can, it will
+    return a $(D ref) to the $(D range element), otherwise it will return
+    a copy.
+ */
+auto ref choice(Range, RandomGen=Random)(auto ref Range range,
+                                         ref RandomGen urng = rndGen)
+    if (isRandomAccessRange!Range && hasLength!Range && isUniformRNG!RandomGen)
+{
+    import std.conv : to, text;
+    import std.exception : enforce;
+
+    enforce(range.length > 0,
+            text("std.random.choice(): invalid Range supplied. Range cannot be empty"));
```

Comment:
>__Your__ 
Unless  this is dedicated to any other participant I'd say that my opinion does not change. To my mind such a small function not being `@nogc`  is a problem (fundamentaly it's nogc but  enforce waste it). 

If you want to keep `enforce` you'll need to wait another review that goes in your direction.